integration-tests/api/__tests__/store/products.js

```diff
@@ -449,6 +453,112 @@ describe("/store/products", () => {
         )
       }
     })
+
+    describe("Product Category filtering", () => {
+      let categoryWithProduct, categoryWithoutProduct, nestedCategoryWithProduct, nested2CategoryWithProduct
+      const nestedCategoryWithProductId = "nested-category-with-product-id"
+      const nested2CategoryWithProductId = "nested2-category-with-product-id"
+      const categoryWithProductId = "category-with-product-id"
+      const categoryWithoutProductId = "category-without-product-id"
+
+      beforeEach(async () => {
+        const manager = dbConnection.manager
+        categoryWithProduct = await simpleProductCategoryFactory(
+          dbConnection,
+          {
+            id: categoryWithProductId,
+            name: "category with Product",
+            products: [{ id: testProductId }],
+          }
+        )
+
+        nestedCategoryWithProduct = await simpleProductCategoryFactory(
+          dbConnection,
+          {
+            id: nestedCategoryWithProductId,
+            name: "nested category with Product1",
+            parent_category: categoryWithProduct,
+            products: [{ id: testProductId1 }],
+          }
+        )
+
+        nested2CategoryWithProduct = await simpleProductCategoryFactory(
+          dbConnection,
+          {
+            id: nested2CategoryWithProductId,
+            name: "nested2 category with Product1",
+            parent_category: nestedCategoryWithProduct,
+            products: [{ id: testProductFilteringId1 }],
+          }
+        )
+
+        categoryWithoutProduct = await simpleProductCategoryFactory(
+          dbConnection,
+          {
+            id: categoryWithoutProductId,
+            name: "category without product",
+          }
+        )
+      })
+
+      it("returns a list of products in product category without category children", async () => {
+        const api = useApi()
+        const params = `category_id[]=${categoryWithProductId}`
```

Comment:
question: is the `[]` needed as a suffix to `category_id` in order for the parameters to be handle properly?

Reply:
question: I think it might be overkill but asking anyway. Should we add a test where `category_children` is set to a "falsey" value?

Reply:
> is the [] needed as a suffix to category_id in order for the parameters to be handle properly

Express requires `[]` to parse it as an array. We could add `string | string[]` to allow for a single id as well - @riqwan wdyt?

Reply:
> question: I think it might be overkill but asking anyway. Should we add a test where `category_children` is set to a "falsey" value?

Added a test for it. 





> We could add `string | string[]` to allow for a single id as well - @riqwan wdyt?

Other filters like sales channels, price lists, etc only follow an array pattern. It would be weird for only categories to allow a string or array. I'd vote for following the same pattern for this as well.